src/main/scala/temple/generate/target/openapi/OpenAPIBuilder.scala

```diff
@@ -0,0 +1,129 @@
+    400 -> OpenAPIBuilder.generateError("Invalid request", "Invalid request parameters: name"),
+    404 -> OpenAPIBuilder.generateError("ID not found", "Object not found with ID 1"),
+    500 -> OpenAPIBuilder.generateError(
+      "The server encountered an error while serving this request",
+      "Unable to reach user service: connection timeout",
+    ),
+  )
+
+  private val paths = mutable.Map[String, mutable.Map[HTTPVerb, Path]]()
+
+  private def path(url: String): mutable.Map[HTTPVerb, Path] =
+    paths.getOrElseUpdate(url, mutable.Map())
+
+  private def isServerAttribute(attribute: Attribute): Boolean = attribute.accessAnnotation contains Annotation.Server
+
+  private def isClientAttribute(attribute: Attribute): Boolean =
+    attribute.accessAnnotation.isEmpty || (attribute.accessAnnotation contains Annotation.Client)
+
+  private def attributeToOpenAPIType(attribute: Attribute): OpenAPISimpleType = attribute.attributeType match {
+    case BoolType     => OpenAPISimpleType("boolean")
+    case DateType     => OpenAPISimpleType("string", "date")
+    case DateTimeType => OpenAPISimpleType("string", "date-time")
+    case TimeType     => OpenAPISimpleType("string", "time")
+    case BlobType(size) =>
+      val maxLength = size.map("maxLength" -> _.asJson)
+      OpenAPISimpleType("string", Seq(maxLength).flatten: _*)
+    case StringType(max, min) =>
+      val minLength = min.map("minLength" -> _.asJson)
+      val maxLength = max.map("maxLength" -> _.asJson)
+      OpenAPISimpleType("string", Seq(minLength, maxLength).flatten: _*)
+    case IntType(max, min, precision) =>
+      val minimum = min.map("minimum" -> _.asJson)
+      val maximum = max.map("maximum" -> _.asJson)
+      OpenAPISimpleType("number", if (precision > 4) "int64" else "int32", Seq(minimum, maximum).flatten: _*)
+    case FloatType(max, min, precision) =>
+      val minimum = min.map("minimum" -> _.asJson)
+      val maximum = max.map("maximum" -> _.asJson)
+      OpenAPISimpleType("number", if (precision > 4) "double" else "float", Seq(minimum, maximum).flatten: _*)
+    case ForeignKey(references) =>
+      OpenAPISimpleType("number", "int32", "description" -> s"Reference to $references ID".asJson)
+  }
+
+  private def generateItemType(attributes: Map[String, Attribute]): OpenAPIObject = OpenAPIObject(
+    attributes.iterator
+      .filter { case _ -> attribute => !isServerAttribute(attribute) }
+      .map { case str -> attribute => str -> attributeToOpenAPIType(attribute) }
+      .to(attributes.mapFactory),
+  )
+
+  private def jsonContent(mediaTypeObject: MediaTypeObject) = Map("application/json" -> mediaTypeObject)
+
+  def addPaths(service: Service): this.type = {
+    val lowerName       = service.name.toLowerCase
+    val capitalizedName = service.name.capitalize
+    val tags            = Seq(capitalizedName)
+    service.endpoints.foreach {
+      case ReadAll =>
+        path(s"/$lowerName/all") += HTTPVerb.Get -> Path(
```

Comment:
This reads a little confusingly, you're calling a function and then adding to it? Took me a while to realise that this will implicitly update the `paths` variable